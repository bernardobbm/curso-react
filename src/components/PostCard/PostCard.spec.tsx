import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { PostCard } from '.';

const mockProps = {
	title: 'postcard',
	body: 'postcard body',
	cover: 'img/img.png',
	id: 1,
};

describe('<PostCard />', () => {
	it('should render PostCard correctly', () => {
		render(<PostCard {...mockProps} />);

		expect(screen.getByRole('img', { name: 'postcard' })).toHaveAttribute('src', 'img/img.png');
		expect(screen.getByRole('heading', { name: 'postcard' })).toBeInTheDocument();
		expect(screen.getByText('postcard body')).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		const { container } = render(<PostCard {...mockProps} />);

		expect(container.firstChild).toMatchSnapshot();
	});
});
