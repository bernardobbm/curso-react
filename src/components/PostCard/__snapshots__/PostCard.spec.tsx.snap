// Vitest Snapshot v1

exports[`<PostCard /> > should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="postcard"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      postcard
    </h2>
    <p>
      postcard body
    </p>
  </div>
</div>
`;
